### PR TITLE
fix(ChildrenWithAge): sync summary and edit container

### PR DESCRIPTION
**Pull request by tujoworker**
This PR is based on #3949 

It adds also support for:

- spacing props
- showEmpty prop

### CODE REVIEW

**Review by langz — APPROVED** [review]
Looks good.
Perhaps we should first merge the following PR https://github.com/dnbexperience/eufemia/pull/3946 to main, then rebase this PR https://github.com/dnbexperience/eufemia/pull/3950 with main, and then add docs to properties for the `showEmpty` property in this PR https://github.com/dnbexperience/eufemia/pull/3950?

**Comment by langz** [comment]
> Looks good. Perhaps we should first merge the following PR #3946 to main(I think that PR is ready for merging), then rebase this PR #3950 with main, and then add docs to the properties tab for the `showEmpty` property in this PR #3950?

PR #3946 is now merged to main 💯

**Comment by langz** [comment]
Tested now, it looks good 🏁 
It's sort of breaking, as we change the default/basic behavior to not showing empty values, but since blocks is sort of beta[(Note: If you are interested in using blocks, please get in touch with us.)](https://eufemia.dnb.no/uilib/extensions/forms/blocks/info/#description), I think it's safe to do so .

File: packages/dnb-eufemia/src/extensions/forms/blocks/ChildrenWithAge/ChildrenWithAge.tsx [hunk 1]
```diff
@@ -48,6 +68,11 @@ function EditContent({ enableAdditionalQuestions }: Props) {
         errorMessages={{
           required: tr.ChildrenWithAge.hasChildren.required,
         }}
+        onChange={(value) => {
+          if (value === false) {
+            update('/countChildren', 0)
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/blocks/ChildrenWithAge/ChildrenWithAge.tsx:73]: Do we need a unit test for this, or maybe it's good enough as it is? 👍

Comment (tujoworker) [reply packages/dnb-eufemia/src/extensions/forms/blocks/ChildrenWithAge/ChildrenWithAge.tsx:73]: Yeah, we can add a test for this case 👍

File: packages/dnb-eufemia/src/extensions/forms/blocks/ChildrenWithAge/__tests__/ChildrenWithAge.screenshot.test.ts [hunk 2]
```diff
@@ -44,6 +44,22 @@ describe.each(['ui', 'sbanken'])('ChildrenWithAge for %s', (themeName) => {
...
+
+  it('have to match field and value when showEmpty', async () => {
+    const screenshot = await makeScreenshot({
+      selector:
+        '[data-visual-test="children-with-age-summary-show-empty"]',
+    })
+    expect(screenshot).toMatchImageSnapshot()
+  })
+
+  it('have to match field and value when showEmpty is false', async () => {
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/blocks/ChildrenWithAge/__tests__/ChildrenWithAge.screenshot.test.ts:56]: We could potentially remove this screenshot test, and the example, as it's sort of handled in the test "have to match field and value when no children" as well. But here we get to see how `showEmpty` is `false` looks directly. I think this test is not directly needed, but no harm(except for a tad bit longer test times)...

Comment (langz) [reply packages/dnb-eufemia/src/extensions/forms/blocks/ChildrenWithAge/__tests__/ChildrenWithAge.screenshot.test.ts]: I removed this now

**Pull request merged**